src/site/HHVMDocumentationSite.php

```diff
@@ -10,84 +10,54 @@
  */
 
 use namespace Facebook\HackRouter;
-use type Facebook\Experimental\Http\Message\{
-  ResponseInterface,
-  ServerRequestInterface,
-};
-use namespace HH\Lib\{File, IO, Math};
+use namespace Nuxed\Contract\Http\{Message, Server};
```

Comment:
Is this required? Would extremely strongly prefer to stick to the existing interfaces.

Reply:
@nuxed interfaces don't extend current facebook interfaces, as i mentioned in slack, there's quite few issues with facebook interfaces, that's why @Nuxed doesn't use them.

Reply:
P.S `Nuxed\Contract` only contains abstractions ( interfaces, traits, enums ..etc), not the implementation itself.

facebook interfaces:

- has only 1 current stable implementation
- uses `KeyedContainer<arraykey, mixed>` for inputs
- uses write only handles for responses
- not standard way to emit responses

nuxed contracts:

- has only 1 current stable implementation
- uses `KeyedContainer<string, string>` for inputs
- supports cookies
- supports session/falsh
- the available implementation offers tons of helper methods to deal with cache, file responses ... etc

Reply:
Yeah ,that mostly feels appropriate for a framework :)

> read/write responses

I'd expect most implementations to provide a read/write (or a pipe) by default, however forcing read/write in the type system bans 'streaming' responses

Almost at the point in HSL IO where updating the interfaces is next - will discuss further there, but I'm leaning towards 'responses should return generic `<T>`', and it's up to the framework - or emitter - to decide what's acceptable

For example, at FB, a 'response body' is an XHP element - for pages, this boils down to the doctype one, for other stuff, it might be a `<myframework:file-response />` which deals with streaming responses - i.e. instead of passing around the place to write the response, make it so the response actually gets returned

Reply:
> forcing read/write in the type system bans 'streaming' responses

how so? currently nuxed/http-emitter supports streaming response :)

Reply:
> For example, at FB, a 'response body' is an XHP element - for pages, this boils down to the doctype one, for other stuff, it might be a <myframework:file-response />

this somehow doesn't feel right to me, XHP is always associated with XML documents or templating in general for me, returning responses that contain only data ( e.g `json` ), by using XHP, seems wrong, even if it works.

Reply:
If it's generic in the interfaces, up to the framework/site - e.g. `XHPResponse implements MyFrameworkResponse`, `JSONResponse implements MyFrameworkResponse`

Reply:
in that case, we would need to send the response from the response implementation, as it might differ, that's not something I want to do.

as I said, read+write doesn't bring any issues, its also possible to pass a handle that implements read+write but throws on write, writing to the response outside the handler ( called controller here ) is possible in a middleware, to modify the response.

Reply:
Read/Write/Seekable: bans directly writing to request_output without buffering.

Always-throw implementations of part of it are removing the safety that the types are meant to provide.

Also, messaged you some more context on why I strongly want to continue using the interfaces.